BUI_Automation.lua

```diff
@@ -576,21 +466,24 @@ function BUI.DailyPledges()
 	if GetCVar("LastRealm")~="NA Megaserver" then offset=1517454000 end
 	local day=math.floor(GetDiffBetweenTimeStamps(GetTimeStamp(),offset)/86400)
 	d("Daily pledges:")
-	for npc=1,3 do
-		local dp=DailyPledges[npc]
-		local n=1+(day+dp.shift)%#dp
-		local pledge=dp[n][BUI.language] or dp[n].en
+	for zoneId, v in pairs(DungeonIndex) do
+		local length = 12
+		local shift = 0
+		if v.npc == "Urgarlag" then -- Override Length if Urg - NEEDS TO BE UPDATED EACH DUNGEON ADDITON
+			length = 32 -- Increment When New Dungeons Added - This is a count of how many quests Urgarlag gives out for daily pledges
+			shift = 27  -- Adjust as necessary with Dungeon Additions - This is an offset that needs to be set manually to make Urgarlag quests show in the correct order. The value can be from 1 through the max number of pledge quests Urgarlag gives
+		end
+		local n=1+(day+shift)%length
+		local pledge=GetZoneNameById(zoneId)
 
-		local quest=""
-		if pledge then
-			local text=pledge:lower()
---			d("DailyPledges => " .. text)
---			d(Pledges[text])
-			if Pledges[text]==false then quest=" |c3388EE- "..BUI.Loc("UndauntedQuest").."|r" end
---			d(text)
+		if (BUI.language == "es") then
+			pledge = pledge:gsub('%^.*', '')
 		end
 
-		d("["..npc.."] "..tostring(pledge)..quest)
+		if n == v.index then
+			d("["..v.npc.."] "..pledge)
```

Comment:
This would an end user visible change. Previously when typing /daily it would say "[0] Dungeon Name", now it will say "[Urgarlag] Dungeon Name" where "Urgarlag" is a hard coded English value. Ideally we should track down the in game variable.